### DISCUSSION TITLE
Web SSH

### DISCUSSION THREAD

**Discussion by KongNan** [nautobot/nautobot#5536] [Plugin (App) Ideas & Development]
### As ...

Nelly - Network Engineer

### I want ...

Can nautobot provide a web ssh for devices such as switches? Currently the nautobot secrets provider is already able to get usernames and passwords from the vault, so providing a web ssh would be very convenient for network administrators and would better promote the use of nautobot.

### So that ...

Some network administrators lack interest in managing nautobot, probably because populating the data is tedious, but the benefits are not obvious enough for them. 

### I know this is done when...

1、On the device page for a specific type of device, an entry point (e.g., button or shortcut, etc.) is added for the corresponding device that can be triggered as an action, and a Web SSH window (either standalone or integrated) pops up when clicked by the network administrator to connect to that device.

2、Ideally, it would be better to read the password from the vault via a secrets provider, which is more convenient and reduces the password hassle and the corresponding security risk.

### Optional - Feature groups this request pertains to.

- [ ] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [X] Organization
- [X] Plugins (and other Extensibility)
- [X] Security (Secrets, etc)
- [ ] Image Management
- [X] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

### Database Changes

NO

**Comment by glennmatthews** [comment]
Thanks for the request! This would be a great candidate for an App, in my opinion, rather than a core feature.
